phpBB/includes/acp/acp_database.php

```diff
@@ -98,50 +101,105 @@ function main($id, $mode)
 
 						$filename = 'backup_' . $time . '_' . unique_id();
 
-						$extractor = $phpbb_container->get('dbal.extractor');
-						$extractor->init_extractor($format, $filename, $time, $download, $store);
+						try
+						{
+							$extractor = $phpbb_container->get('dbal.extractor');
+							$extractor->init_extractor($format, $filename, $time, $download, $store);
 
-						$extractor->write_start($table_prefix);
+							$extractor->write_start($table_prefix);
 
-						foreach ($table as $table_name)
-						{
-							// Get the table structure
-							if ($structure)
+							foreach ($table as $table_name)
 							{
-								$extractor->write_table($table_name);
+								// Get the table structure
+								if ($structure)
+								{
+									$extractor->write_table($table_name);
+								}
+								else
+								{
+									// We might wanna empty out all that junk :D
+									switch ($db->get_sql_layer())
+									{
+										case 'sqlite3':
+											$extractor->flush('DELETE FROM ' . $table_name . ";\n");
+										break;
+
+										case 'mssql_odbc':
+										case 'mssqlnative':
+											$extractor->flush('TRUNCATE TABLE ' . $table_name . "GO\n");
+										break;
+
+										case 'oracle':
+											$extractor->flush('TRUNCATE TABLE ' . $table_name . "/\n");
+										break;
+
+										default:
+											$extractor->flush('TRUNCATE TABLE ' . $table_name . ";\n");
+										break;
+									}
+								}
+
+								// Data
```

Comment:
Kind of useless comment. Either state something like `Write schema data if it exists` (which one can actually deduct from the code or remove it (which I think I prefer in this case).